### PR TITLE
chore(package.json): upgrade react

### LINKED ISSUE #209: upgrade react

### PR DESCRIPTION
**Pull request by adnasa**
well... I guess I overestimated the upgrade

- upgrades `react`
- **bug**: uses `*InputRef` over `ref` prop when attempting to attach a DOM node via intermediate components

closes #209

### CODE REVIEW

File: app/renderer/src/containers/query-field-container/query-field-container.js [hunk 1]
```diff
@@ -6,54 +6,47 @@ import { ThemeSchema } from '../../schema';
...
+  };
 
-  attach = c => {
-    this.queryField = c;
+  attachQueryField = queryField => {
+    this.queryField = queryField;
   };
 
   render() {
-    const { q, theme } = this.props;
```
Comment (adnasa) [inline app/renderer/src/containers/query-field-container/query-field-container.js:50]: needs to be followed up with https://github.com/DextApp/dext/issues/210

File: app/renderer/src/containers/query-field-container/query-field-container.js [hunk 2]
```diff
@@ -6,54 +6,47 @@ import { ThemeSchema } from '../../schema';
...
-    // resets the querty and blurs the input field when the window is hidden
     ipcRenderer.on(IPC_WINDOW_HIDE, () => {
-      this.props.onReset && this.props.onReset();
-      self.blur();
+      if (this.props.onReset) this.props.onReset();
+      this.blur();
     });
   }
 
-  focus() {
```
Comment (adnasa) [inline app/renderer/src/containers/query-field-container/query-field-container.js:28]: we should start preferring property initializers from here on

Comment (vutran) [reply app/renderer/src/containers/query-field-container/query-field-container.js:28]: +1

File: package.json [hunk 3]
```diff
@@ -86,11 +85,12 @@
...
-    "react": "^15.3.1",
-    "react-dom": "^15.3.1",
-    "react-redux": "^5.0.2",
-    "redux": "^3.5.2",
-    "redux-saga": "^0.14.3",
+    "react": "16.3.2",
+    "react-dom": "16.3.2",
+    "react-redux": "5.0.7",
+    "react-test-renderer": "16.3.2",
+    "redux": "3.7.2",
```
Comment (vutran) [inline package.json:92]: Is `redux` + redux friends necessary for the `react` upgrade. If so, so be it since they'll be gone anyways.

**Review by vutran — APPROVED** [review]

File: package.json (hunk 3 above)
Comment (adnasa) [reply package.json:92]: it's better than having old version dangling around for now :)

**Pull request merged**